includes/Features/Abilities_Explorer/Ability_Handler.php

```diff
@@ -0,0 +1,331 @@
+			return null;
+		}
+	}
+
+	/**
+	 * Format abilities array.
+	 *
+	 * @since 0.1.0
+	 *
+	 * @param array $abilities Raw abilities array (WP_Ability objects).
+	 * @return array Formatted abilities.
+	 */
+	private static function format_abilities( $abilities ) {
+		if ( empty( $abilities ) || ! is_array( $abilities ) ) {
+			return array();
+		}
+
+		$formatted = array();
+
+		foreach ( $abilities as $ability ) {
+			$formatted[] = self::format_single_ability( $ability );
+		}
+
+		return $formatted;
+	}
+
+	/**
+	 * Format a single ability.
+	 *
+	 * @since 0.1.0
+	 *
+	 * @param \WP_Ability $ability Ability object.
+	 * @return array Formatted ability data.
+	 */
+	private static function format_single_ability( $ability ) {
+		if ( ! is_object( $ability ) || ! method_exists( $ability, 'get_name' ) ) {
+			return array();
+		}
+
+		$name = $ability->get_name();
+		$meta = method_exists( $ability, 'get_meta' ) ? $ability->get_meta() : array();
+
+		return array(
+			'slug'         => $name,
+			'name'         => $ability->get_label(),
+			'description'  => $ability->get_description(),
+			'provider'     => self::detect_provider( $name, $meta ),
+			'input_schema' => $ability->get_input_schema(),
+			'output_schema' => $ability->get_output_schema(),
+			'raw_data'     => array(
+				'name'         => $name,
+				'label'        => $ability->get_label(),
+				'description'  => $ability->get_description(),
+				'input_schema' => $ability->get_input_schema(),
+				'output_schema' => $ability->get_output_schema(),
+				'meta'         => $meta,
+			),
+		);
+	}
+
+	/**
+	 * Detect ability provider (Core, Plugin, or Theme).
+	 *
+	 * @since 0.1.0
+	 *
+	 * @param string $name Ability name (slug).
+	 * @param array  $meta Ability metadata.
+	 * @return string Provider type.
+	 */
+	private static function detect_provider( $name, $meta ) {
+		// Check if provider is explicitly set in meta
+		if ( isset( $meta['provider'] ) ) {
+			return $meta['provider'];
+		}
+
+		// Detect based on name prefix (namespace/ability format)
+		$parts = explode( '/', $name );
+		if ( count( $parts ) === 2 ) {
+			$namespace = $parts[0];
+
+			// WordPress core abilities
+			if ( in_array( $namespace, array( 'wordpress', 'wp', 'core' ), true ) ) {
+				return 'Core';
+			}
+
+			// Check if namespace matches active theme
+			if ( $namespace === get_stylesheet() || $namespace === get_template() ) {
+				return 'Theme';
+			}
+		}
+
+		// Default to Plugin
+		return 'Plugin';
+	}
+
+	/**
+	 * Invoke an ability.
+	 *
+	 * @since 0.1.0
+	 *
+	 * @param string $slug  Ability name.
+	 * @param array  $input Input data.
+	 * @return array Result with success status and data/error.
+	 */
+	public static function invoke_ability( $slug, $input = array() ) {
+		if ( ! function_exists( 'wp_get_ability' ) ) {
+			return array(
+				'success' => false,
+				'error'   => 'Abilities API not available',
+			);
+		}
+
+		try {
```

Comment:
Does this need to be inside a try catch block? Usually a try catch block is used when we don't know what to expect (for example an external API call that we have no control over). AFAIK calling wp_get_ability, $ability->get_input_schema() and $ability->execute() already contains error handling.